py-polars/tests/unit/operations/map/test_inefficient_map_warning.py

```diff
@@ -406,3 +407,13 @@ def test_expr_exact_warning_message() -> None:
         df.select(pl.col("a").map_elements(lambda x: x + 1))
 
     assert len(warnings) == 1
+
+
+def test_partial_functions_13523() -> None:
+    def plus(value, amount: int):  # type: ignore[no-untyped-def]
+        return value + amount
+
+    data = {"a": [1, 2], "b": [3, 4]}
+    df = pl.DataFrame(data)
+    # should not warn
+    _ = df["a"].map_elements(partial(plus, amount=1))
```

Comment:
Let's actually make sure this doesn't warn?

```suggestion
def test_partial_functions_13523(recwarn: Any) -> None:
    def plus(value, amount: int):  # type: ignore[no-untyped-def]
        return value + amount

    data = {"a": [1, 2], "b": [3, 4]}
    df = pl.DataFrame(data)
    _ = df["a"].map_elements(partial(plus, amount=1))
    assert len(recwarn) == 0
```

Reply:
I was too late 😄 but it's fine, don't worry about it.